DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/mmhost/README.md

```diff
@@ -0,0 +1,372 @@
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+
+These parameters can be used to improve the performance of `Example 1` by ensuring that each pointer points to data in a dedicated Avalon memory-mapped agent memory, like this:
+
+![](assets/dedicated_avhost.svg)
+
+#### Example 3: A kernel with dedicated Avalon memory-mapped host interfaces
+(Code can be found under `part3_hosts`).
+```c++
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+constexpr int kBL3 = 3;
+constexpr int kAlignment = 4;
+
+struct MultiMMIP {
+  // Each annotated pointer is configured with a unique `buffer_location`,
+  // resulting in three unique Avalon memory-mapped host interfaces.
+  using XProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL1>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_read});
+  using YProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL2>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_read});
+  using ZProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL3>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_write});
+
+  sycl::ext::oneapi::experimental::annotated_arg<int *, XProps> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, YProps> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ZProps> z;
+
+  int size;
+
+  void operator()() const {
+#pragma unroll 4
+    for (int i = 0; i < size; i++) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+```
+
+### Interfacing with off-chip memory
+
+If the input and output vectors are too large for on-chip memory, larger off-chip memories can be used. Consider the parameterization of a system with the following off-chip memory interfaces:
+
+* Two banks of DDR SDRAM
+* Data bus of 256 bits
+* bursts of up to 8 requests
+
+![](assets/ddr_avhost.svg)
+
+The available memory bandwidth can be better used by coalescing the 32-bit wide load-store units into wider 256-bit wide load-store units to match the memory interface.
+
+#### Example 4: A kernel that interfaces with two off-chip memories
```

Comment:
We should mention how specifying `alignment` property allows the compiler to assume the specified `alignment` and infer an optimized LSU. Without this property, compiler must infer a non-aligned LSU that has worse qor because it requires additional logic to handle potential unaligned accesses.

We should also state how when the `alignment` property is specified on the kernel argument, the user **must** also specify the same alignment to the SYCL runtime using the `aligned_alloc_shared` API as shown in the codesample otherwise the simulation run may have functional failures.